ParkingZoneApp/Areas/Admin/Controllers/ParkingZoneController.cs

```diff
@@ -6,156 +6,112 @@
 
-        // GET: ParkingZone/Create
+        // GET: Admin/ParkingZone/Create
         public IActionResult Create()
         {
             return View();
         }
 
-        // POST: ParkingZone/Create
-        // To protect from overposting attacks, enable the specific properties you want to bind to.
-        // For more details, see http://go.microsoft.com/fwlink/?LinkId=317598.
         [HttpPost]
         [ValidateAntiForgeryToken]
-        public async Task<IActionResult> Create([Bind("Id,Name,Address,DateOfEstablishment")] ParkingZone parkingZoneModel)
+        public IActionResult Create(ParkingZone parkingZone)
         {
             if (ModelState.IsValid)
             {
-                _context.Add(parkingZoneModel);
-                await _context.SaveChangesAsync();
+                _repository.Insert(parkingZone);
+                _repository.Save();
                 return RedirectToAction(nameof(Index));
             }
-            return View(parkingZoneModel);
+            return View(parkingZone);
         }
 
-        // GET: ParkingZone/Edit/5
-        public async Task<IActionResult> Edit(int? id)
+        // GET: Admin/ParkingZone/Edit/5
+        public IActionResult Edit(int id)
         {
-            if (id == null)
+            var ParkingZoneById = _repository.GetById(id);
+            if (ParkingZoneById is not null)
             {
-                return NotFound();
+                return View(ParkingZoneById);
             }
 
-            var parkingZoneModel = await _context.ParkingZone.FindAsync(id);
-            if (parkingZoneModel == null)
-            {
-                return NotFound();
-            }
-            return View(parkingZoneModel);
+            return NotFound();
         }
 
-        // POST: ParkingZone/Edit/5
-        // To protect from overposting attacks, enable the specific properties you want to bind to.
-        // For more details, see http://go.microsoft.com/fwlink/?LinkId=317598.
         [HttpPost]
         [ValidateAntiForgeryToken]
-        public async Task<IActionResult> Edit(int id, [Bind("Id,Name,Address,DateOfEstablishment")] ParkingZone parkingZoneModel)
+        public IActionResult Edit(int id, ParkingZone parkingZone)
         {
-            if (id != parkingZoneModel.Id)
-            {
-                return NotFound();
-            }
-
             if (ModelState.IsValid)
             {
-                try
-                {
-                    _context.Update(parkingZoneModel);
-                    await _context.SaveChangesAsync();
-                }
-                catch (DbUpdateConcurrencyException)
-                {
-                    if (!ParkingZoneModelExists(parkingZoneModel.Id))
-                    {
-                        return NotFound();
-                    }
-                    else
-                    {
-                        throw;
-                    }
-                }
+                _repository.Update(parkingZone);
```

Comment:
before updating get parkingZone from database and check whether it exists or not with this id